output/include/openthread.h

```diff
@@ -0,0 +1,1701 @@
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup api  API
+ * @brief
+ *   This module includes the application programming interface to the OpenThread stack.
+ *
+ * @{
+ *
+ * @defgroup execution Execution
+ * @defgroup commands Commands
+ * @defgroup config Configuration
+ * @defgroup diags Diagnostics
+ * @defgroup messages Message Buffers
+ * @defgroup ip6 IPv6
+ * @defgroup udp UDP
+ *
+ * @}
+ *
+ */
+
+/**
+ * @defgroup platform  Platform Abstraction
+ * @brief
+ *   This module includes the platform abstraction used by the OpenThread stack.
+ *
+ * @{
+ * @}
+ *
+ */
+
+/**
+ * @defgroup core Core
+ * @brief
+ *   This module includes the core OpenThread stack.
+ *
+ * @{
+ *
+ * @defgroup core-6lowpan 6LoWPAN
+ * @defgroup core-coap CoAP
+ * @defgroup core-ipv6 IPv6
+ * @defgroup core-mac MAC
+ * @defgroup core-mesh-forwarding Mesh Forwarding
+ * @defgroup core-message Message
+ * @defgroup core-mle MLE
+ * @defgroup core-netdata Network Data
+ * @defgroup core-netif Network Interface
+ * @defgroup core-arp RLOC Mapping
+ * @defgroup core-security Security
+ * @defgroup core-tasklet Tasklet
+ * @defgroup core-timer Timer
+ * @defgroup core-udp UDP
+ * @defgroup core-link-quality Link Quality
+ *
+ * @}
+ *
+ */
+
+/**
+ * @addtogroup execution  Execution
+ *
+ * @brief
+ *   This module includes functions that control the Thread stack's execution.
+ *
+ * @{
+ *
+ */
+
+/**
+ * Run the next queued tasklet in OpenThread.
+ * 
+ * @param[in] aContext  The OpenThread context structure.
+ */
+void otProcessNextTasklet(otContext *aContext);
+
+/**
+ * Indicates whether or not OpenThread has tasklets pending.
+ * 
+ * @param[in] aContext  The OpenThread context structure.
+ *
+ * @retval TRUE   If there are tasklets pending.
+ * @retval FALSE  If there are no tasklets pending.
+ */
+bool otAreTaskletsPending(otContext *aContext);
+
+/**
+ * OpenThread calls this function when the tasklet queue transitions from empty to non-empty.
+ * 
+ * @param[in] aContext  The OpenThread context structure.
+ *
+ */
+extern void otSignalTaskletPending(otContext *aContext);
+
+/**
+ * @}
+ *
+ */
+
+/**
+ * @addtogroup commands  Commands
+ *
+ * @brief
+ *   This module includes functions for OpenThread commands.
+ *
+ * @{
+ *
+ */
+
+/**
+ * This function initializes the OpenThread library.
+ *
+ * This function initializes OpenThread and prepares it for subsequent OpenThread API calls.  This function must be
+ * called before any other calls to OpenThread.
+ * 
+ * @param[in]    aContextBuffer      The buffer for OpenThread to use for allocating the otContext structure.
+ * @param[inout] aContextBufferSize  On input, the size of aContextBuffer. On output, if not enough space for otContext, 
+                                     the number of bytes required for otContext.
+ *
+ * @retval otContext*  The new OpenThread context structure.
+ *
+ */
+otContext *otEnable(void *aContextBuffer, uint64_t *aContextBufferSize);
+
```

Comment:
I think it might be good to bring back otInit to allocate, and initialize the otContext, and make otEnable/otDisable just manage the mEnabled state. What do you guys think?